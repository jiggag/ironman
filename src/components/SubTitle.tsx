import React, { memo } from 'react';
import { Text, StyleSheet } from 'react-native';

interface SubTitleType {
  title: string;
  style?: object;
}

const SubTitle = memo(({ title, style = {} }: SubTitleType) => <Text style={[styles.subTitle, style]}>{title}</Text>);

export default SubTitle;

const styles = StyleSheet.create({
  subTitle: {
    fontSize: 16,
    fontWeight: '500',
    textDecorationLine: 'underline',
    marginVertical: 4,
  },
});
