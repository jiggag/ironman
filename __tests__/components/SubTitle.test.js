import React from 'react';
import renderer from 'react-test-renderer';
import SubTitle from '../../src/components/SubTitle';

it('Renderer Component: SubTitle', () => {
  const component = renderer.create(
    <SubTitle title="서브타이틀" />
  )
  let subTitle = component.toJSON();
  expect(subTitle).toMatchSnapshot();
});