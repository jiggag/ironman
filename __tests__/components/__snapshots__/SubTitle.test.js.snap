// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer Component: SubTitle 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 16,
        "fontWeight": "500",
        "marginVertical": 4,
        "textDecorationLine": "underline",
      },
      Object {},
    ]
  }
>
  서브타이틀
</Text>
`;
